packages/shared/client/ReplayClient.ts

```diff
@@ -417,10 +418,17 @@ export class ReplayClient implements ReplayClientInterface {
     const newSourceListener = (source: Source) => {
       sources.push(source);
     };
+    const newSourcesListener = ({ sources }: newSources) => {
+      for (const source of sources) {
+        sources.push(source);
+      }
+    };
 
     client.Debugger.addNewSourceListener(newSourceListener);
+    client.Debugger.addNewSourcesListener(newSourcesListener);
```

Comment:
What prevents us from adding every source twice? Is the expectation here that _only_ one of these two events will fire for any given session?

Reply:
Yeah the API wouldn't emit both, just one or the other. I'm going to delete the other listener as soon as things is deployed, I just have both in this transition period.

Reply:
@loganfsmyth note that the routines logic depends on fetching sources as well

Reply:
Yup that should all be updated with the backend conversion https://github.com/replayio/backend/pull/7961

Reply:
Woops wrong PR, this one: https://github.com/replayio/backend/pull/7958